emmet-core/emmet/core/qchem/calculation.py

```diff
@@ -0,0 +1,437 @@
+    )
+
+    frequencies: Dict[str, Any] = Field(
+        None,
+        description="Calculated frequency modes if the job type is freq or frequency",
+    )
+
+    frequency_modes: Union(List, str) = Field(
+        None, description="The list of calculated frequency mode vectors"
+    )
+
+    final_energy: Union(str, float) = Field(
+        None,
+        description="The final energy of the molecule after the calculation is complete",
+    )
+
+    enthalpy: Union(str, float) = Field(
+        None,
+        description="The total enthalpy correction if a frequency calculation has been performed",
+    )
+
+    entropy: Union(str, float) = Field(
+        None,
+        description="The total entropy of the system if a frequency calculation has been performed",
+    )
+
+    scan_energies: Dict[str, Any] = Field(
+        None,
+        description="A dictionary of the scan energies with their respective parameters",
+    )
+
+    scan_geometries: Dict[str, Any] = Field(
+        None, description="optimized geometry of the molecules after the geometric scan"
+    )
+
+    scan_molecules: Dict[str, Any] = Field(
+        None,
+        description="The constructed pymatgen molecules from the optimized scan geometries",
+    )
+
+    pcm_gradients: Union(Dict[str, Any], np.ndarray) = Field(
+        None,
+        description="The parsed total gradients after adding the PCM contributions.",
+    )
+
+    cds_gradients: Union(Dict[str, Any], np.ndarray) = Field(
+        None, description="The parsed CDS gradients."
+    )
+
+    dipoles: Dict[str, Any] = Field(
+        None, description="The associated dipoles for Mulliken/RESP/ESP charges"
+    )
+
+    gap_info: Dict[str, Any] = Field(
+        None, description="The Kohn-Sham HOMO-LUMO gaps and associated Eigenvalues"
+    )
+
+    @classmethod
+    def from_qcoutput(cls, qcoutput: QCOutput) -> "CalculationOutput":
+        """
+        Create a QChem output document from a QCOutput object.
+
+        Parameters
+        ----------
+        qcoutput
+            A QCOutput object.
+
+        Returns
+        --------
+        CalculationOutput
+            The output document.
+        """
+
+        return cls(
+            optimized_molecule=qcoutput.data["molecule_from_optimized_geometry"],
+            mulliken=qcoutput.data["Mulliken"][-1],
+            esp=qcoutput.data["ESP"][-1],
+            resp=qcoutput.data["RESP"][-1],
+            nbo_data=qcoutput.data["nbo_data"],
+            frequencies=qcoutput.data["frequencies"],
+            frequency_modes=qcoutput.data.get(["frequency_mode_vectors"], []),
+            final_energy=qcoutput.data["final_energy"],
+            enthalpy=qcoutput.data["enthalpy"],
+            entropy=qcoutput.data["entropy"],
+            scan_energies=qcoutput.data.get(["scan_energies"]),
+            scan_geometries=qcoutput.data["optimized_geometries"]
+            if qcoutput.data["scan_energies"]
+            else None,
```

Comment:
I don't know what the current preferred style is, but I'd rather spend a few extra lines and have this logic separate.